### ISSUE TITLE
Import CSV 2.0

### ISSUE THREAD

**Issue by txau** [huridocs/uwazi#2310]
- [x] Add an icon in uploads for importing CSV file
- [x] Halt on first error and provides feedback
- [x] Add support for multiple languages (add id and lang columns)
- [x] Support updates when Uwazi ids are provided
- [x] Support file uploads (upload a zip)
- [ ] validate zip has only one csv
- [x] When I don't have a lang, I use the default lang to create the records
- [ ] ~~Add an icon for importing thesauri in a thesauri~~

**Comment by RafaPolit** [comment]
- [ ] Be able to delete file once assigned (removing the column has no effect, assigning empty value to the column has no effect either)

**Comment by RafaPolit** [comment]
This is working, we have some general UX questions that we have included in #2353

**Comment by daneryl** [comment]
@RafaPolit @txau 

> The UI, we have some general UX questions that we have included in #2353, or maybe everything should be consolidated here?

lets finish the the more tech side here and continue with ux things in the other one.

> Be able to delete file once assigned (removing the column has no effect, assigning empty value to the column has no effect either)

should a blank value on the file column delete the file then ?

**Comment by daneryl** [comment]
@txau thinking about this

>  validate zip has only one csv

what if they want to add a csv as an attachment, should we some kind of naming convention to specify which csv to use for the import  ?

**Comment by txau** [comment]
@daneryl good call. It totally makes sense.

**Comment by daneryl** [comment]
@txau, so do nothing about it until we implement attachments sounds good ?

**Comment by txau** [comment]
@daneryl since you are going to check nevertheless for the presence of a CSV, check for a particular CSV name, ie. import.csv.

**Comment by habbes** [comment]
@danery when you have a .zip file with multiple files, including multiple .csv files, it correctly imports the `import.csv` file.

If there's no `import.csv` in the zip folder, this notifcation is displayed:
<img width="395" alt="Screenshot 2019-07-18 at 14 42 32" src="https://user-images.githubusercontent.com/8460169/61455022-1f1b7a00-a96b-11e9-8e90-656f94e53eb0.png">

It stays on the sidebar indefinitely. But you can close it.

The notification goes on the top bar when you go to the library page:
<img width="1679" alt="Screenshot 2019-07-18 at 14 46 19" src="https://user-images.githubusercontent.com/8460169/61455082-407c6600-a96b-11e9-88de-4bc8b1107a3c.png">

I can't see any way to close it. It stays there indefinitely, until you refresh the page.

**Comment by daneryl** [comment]
@habbes ok, nice one ... will look into it thx !

**Comment by daneryl** [comment]
@habbes the fix is ready on the PR, error handling was missing on the endpoint, ready for QA

**Comment by habbes** [comment]
@daneryl alright

And regarding this:

> what if they want to add a csv as an attachment, should we some kind of naming convention to specify which csv to use for the import ?

How do I add attachments in the zip for imports?

**Comment by daneryl** [comment]
@habbes attachments support is not implemented yet, this was just to prevent this problem in the future.

**Comment by habbes** [comment]
@daneryl oh alright

**Comment by habbes** [comment]
@daneryl I've tested again and now it seems to work fine.

I found the icon used for the error message to be strange, but I don't think that's a major issue
<img width="402" alt="Screenshot 2019-07-19 at 15 31 12" src="https://user-images.githubusercontent.com/8460169/61535438-f5ca1f00-aa3a-11e9-9f97-9d0dd141fec4.png">

I also tried using a file named `import.csv` that was actually an image and not an a csv file and it showed the following error:
<img width="403" alt="Screenshot 2019-07-19 at 15 34 53" src="https://user-images.githubusercontent.com/8460169/61535482-11cdc080-aa3b-11e9-9ecb-850c60ef1bc3.png">

I guess that's okay as well.

And finally, I created a template with a required field and used imported a csv file that does not have  that field. The import worked without reporting any issues. And the entities were create successfully, without the required field. Of course I don't think this is an issue with the import feature, I just think it means we don't have server-side validation of entities against the template schema.

If the error icon and the handling of non-csv file are note considerable issues, then this feature looks good to me.
